src/main/java/org/testng/internal/DynamicGraph.java

```diff
@@ -50,64 +49,103 @@ public void addNode(T node) {
   /**
    * Add an edge between two nodes.
    */
-  public void addEdge(T from, T... tos) {
-    addEdge(from, Arrays.asList(tos));
+  public void addEdge(int weight, T from, T... tos) {
+    addEdge(weight, from, Arrays.asList(tos));
   }
 
   /**
    * Add an edge between two nodes.
    */
-  public void addEdge(T from, Iterable<T> tos) {
+  public void addEdge(int weight, T from, Iterable<T> tos) {
```

Comment:
> you could add an overload for addEdge() that sets weight to 0 (not sure it's useful yet, not done reviewing everything).

No problem for me, I've already debated about it with me ;) But it will be only used once. 
I think keeping `weight` could be important because the value is important and it will avoid unexpected behavior when the default value is used with configured ones (imagine if you set negative weight for any reason). Then the method user takes the responsibility to use the good weight.

Let me know what you prefer (adding a new method or setting the value at the good place)